src/libraries/Microsoft.PowerFx.Core/Public/Values/PagedRecordValue.cs

```
@@ -0,0 +1,137 @@
+            if (fieldType is TableType && result is TableValue tv)
+            {
+                result = new PagedTableValue(tv, GetNextRecord, fieldType, fieldName, MaxRows);
+            }
+
+            return b;
+        }
+    }
+
+    internal class PagedTableValue : TableValue
+    {
+        public TableValue CurrentTable { get; private set; }
+
+        public Func<FormulaValue> GetNextRecord { get; private set; }
+
+        public FormulaType FieldType { get; private set; }
+
+        public string FieldName { get; private set; }
+
+        public int MaxRows { get; private set; }
+
+        public PagedTableValue(TableValue tableValue, Func<FormulaValue> getNextRecord, FormulaType fieldType, string fieldName, int maxRows)
+            : base(tableValue.Type)
+        {
+            CurrentTable = tableValue;
+            GetNextRecord = getNextRecord;
+            FieldType = fieldType;
+            FieldName = fieldName;
+            MaxRows = maxRows;
+        }
+
+        public override IEnumerable<DValue<RecordValue>> Rows
+        {
+            get
+            {
+                int rowCount = 0;
+
+                // Enumerate rows of the current page
+                foreach (DValue<RecordValue> currentRow in CurrentTable.Rows)
+                {
+                    rowCount++;
+                    if (rowCount > MaxRows)
+                    {
+                        break;
+                    }
+
+                    yield return currentRow;
+                }
+
+                PagedTableValue ptv = this;
+
+                while (ptv != null)
+                {
+                    if (rowCount > MaxRows)
+                    {
+                        break;
+                    }
+
+                    // Get the next page and if an error occurs, let's stop
+                    if (ptv.GetNextRecord() is not RecordValue nextRecord)
+                    {
+                        break;
```

>break; [](http://example.com/codeflow?start=24&length=6)

yield error here?
Concerned we may be swallowing errors